### ISSUE TITLE
Missmatch in number of choices for MCTS AI

### ISSUE THREAD

**Issue by firemind** [magarena/magarena#74]
I've noticed quite a few duels on firemind.ch failing with a java indexOutOfBounds exception. I traced them to two places in MCTSAI.java and saw that there were some asserts that would catch those.

After changing the worker to run magarena with assertions enabled ( java -ea ... ) I'm getting these errors in about 1 out of 30 duels: 

    Exception from controller.runGame: ERROR! Root has no children but there are 8 choices
    java.lang.AssertionError: ERROR! Root has no children but there are 8 choices
      at magic.ai.MCTSAI.findNextEventChoiceResults(MCTSAI.java:160)
      at magic.headless.HeadlessGameController.getAIChoiceResults(HeadlessGameController.java:50)
      at magic.headless.HeadlessGameController.runGame(HeadlessGameController.java:33)
      at magic.FiremindQueueWorker.runDuel(FiremindQueueWorker.java:179)

**Comment by firemind** [comment]
I uploaded the crash.log here https://gist.github.com/firemind/b6d0f017929fb8a9cc3f

**Comment by melvinzhang** [comment]
Thanks for the report. Is this from 1.58 release or from development? If it
is from development, what is the commit hash?

**Comment by firemind** [comment]
It's directly from the master branch. Last commit:

commit ab8cd45ab813f4c8026731d0575dd49979965c4c
Author: melvinzhang <melvinzhang@gmail.com>
Date:   Sat Feb 21 17:23:53 2015 +0800

    support for recognizing mtd(f) in DeckStrCal results


But it's not like this is something new. I remember some of these popping up occasionally ever since I started working on Project Firemind. I'm just now more aware of them because I changed the way I get error reports.
